pennylane/ops/op_math/linear_combination.py

```diff
@@ -0,0 +1,777 @@
+                with 8 stored elements in Compressed Sparse Row format>
+
+        The resulting sparse matrix can be either used directly or transformed into a numpy array:
+
+        >>> H_sparse.toarray()
+        array([[ 1.+0.j  ,  0.+0.j  ,  0.+0.45j,  0.+0.j  ],
+               [ 0.+0.j  , -1.+0.j  ,  0.+0.j  ,  0.-0.45j],
+               [ 0.-0.45j,  0.+0.j  , -1.+0.j  ,  0.+0.j  ],
+               [ 0.+0.j  ,  0.+0.45j,  0.+0.j  ,  1.+0.j  ]])
+        """
+        if wire_order is None:
+            wires = self.wires
+        else:
+            wires = wire_order
+        n = len(wires)
+        matrix = scipy.sparse.csr_matrix((2**n, 2**n), dtype="complex128")
+
+        coeffs = qml.math.toarray(self.data)
+
+        temp_mats = []
+        for coeff, op in zip(coeffs, self.ops):
+            obs = []
+            for o in qml.operation.Tensor(op).obs:
+                if len(o.wires) > 1:
+                    # todo: deal with operations created from multi-qubit operations such as Hermitian
+                    raise ValueError(
+                        f"Can only sparsify LinearCombinations whose constituent observables consist of "
+                        f"(tensor products of) single-qubit operators; got {op}."
+                    )
+                obs.append(o.matrix())
+
+            # Array to store the single-wire observables which will be Kronecker producted together
+            mat = []
+            # i_count tracks the number of consecutive single-wire identity matrices encountered
+            # in order to avoid unnecessary Kronecker products, since I_n x I_m = I_{n+m}
+            i_count = 0
+            for wire_lab in wires:
+                if wire_lab in op.wires:
+                    if i_count > 0:
+                        mat.append(scipy.sparse.eye(2**i_count, format="coo"))
+                    i_count = 0
+                    idx = op.wires.index(wire_lab)
+                    # obs is an array storing the single-wire observables which
+                    # make up the full LinearCombination term
+                    sp_obs = scipy.sparse.coo_matrix(obs[idx])
+                    mat.append(sp_obs)
+                else:
+                    i_count += 1
+
+            if i_count > 0:
+                mat.append(scipy.sparse.eye(2**i_count, format="coo"))
+
+            red_mat = (
+                functools.reduce(lambda i, j: scipy.sparse.kron(i, j, format="coo"), mat) * coeff
+            )
+
+            temp_mats.append(red_mat.tocsr())
+            # Value of 100 arrived at empirically to balance time savings vs memory use. At this point
+            # the `temp_mats` are summed into the final result and the temporary storage array is
+            # cleared.
+            if (len(temp_mats) % 100) == 0:
+                matrix += sum(temp_mats)
+                temp_mats = []
+
+        matrix += sum(temp_mats)
+        return matrix
+
+    def simplify(self):
```

Comment:
Could we also update the `simplify` method to not act in-place? The only reason `Hamiltonian.simplify` does it is that it was created before `Operator.simplify`.

Reply:
Also, this code is really, really slow, so I think we should also update this to follow operator-arithmetic style simplificaiton instead.

Reply:
Yep its on the To-Do above! :)

Reply:
Do we want to continue to support simplify upon initialization, i.e. `qml.LinearCombination(coeffs, ops, simplify=True)`?